reliability/stub-core/src/main/kotlin/com/datadog/android/core/stub/StubInternalLogger.kt

```diff
@@ -12,6 +12,8 @@ import com.datadog.android.core.metrics.TelemetryMetricType
 
 @Suppress("UnsafeThirdPartyFunctionCall")
 internal class StubInternalLogger : InternalLogger {
+
+    var lastMetric: Map<String, Any> = hashMapOf()
     override fun log(
```

Comment:
following approach of capturing all the data in the test, it is probably more useful to record all the events (and then access the necessary using `last()` or `findLast`, etc.), also making this property immutable

```suggestion
    val telemetryEventsWritten = mutableListOf<Map<String, Any>>()

    override fun log(
```

Reply:
updated